File: Data Structures & Algorithms/two-integer-sum/submission-0.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:

        map = {}
        index = 0
        for i in nums:
            if target - i in map: 
                return [map.get(target - i, 0), index]
            else:
                map[i] = index
                index += 1

        return [0, 0]
        

File: Data Structures & Algorithms/daily-temperatures/submission-0.cpp
class Solution {
public:
    vector<int> dailyTemperatures(vector<int>& temperatures) {
        stack<int> temp_index;
        vector<int> result(temperatures.size());

        int index = 0;
        for(int temp : temperatures) {

            if(!temp_index.empty()) {
                int top = temp_index.top();
                while(temperatures[top] < temp) {

                    result[top] = index - top;

                    temp_index.pop();
                    if(temp_index.empty()) break;
                    top = temp_index.top();
                }
            }

            temp_index.push(index);
            ++index;
        }

        return result;
    }
};
